web/assets/js/app.js

```diff
@@ -1,372 +1,344 @@
+        po4_mean: assertSetOrDefault(this.props?.po4_mean, this.props?.po4_mean.toFixed(4), "-"),
+        pbo_mean: assertSetOrDefault(this.props?.pbo_mean, this.props?.pbo_mean.toFixed(4), "-"),
+        pbo_sum: assertSetOrDefault(this.props?.pbo_sum, this.props?.pbo_sum.toFixed(4), "-"),
+        tob_mean: assertSetOrDefault(this.props?.tob_mean, this.props?.tob_mean.toFixed(4), "-"),
+        tob_sum: assertSetOrDefault(this.props?.tob_sum, this.props?.tob_sum.toFixed(4), "-"),
       };
     },
 
-    // initialization
     init() {
-      // create Mapbox map
-      map = new mapboxgl.Map({
+      this.map = new mapboxgl.Map({
         container: "map",
         center: { lng: 42.3586993633067, lat: -18.7128841592949 },
         zoom: 3,
         minZoom: 3,
         maxZoom: 8,
         style: "mapbox://styles/mapbox/light-v11",
-        // style: 'mapbox://styles/mapbox/streets-v11',
         projection: {
           name: "mercator",
         },
       });
 
-      // add zoom controls
-      map.addControl(new mapboxgl.NavigationControl());
+      this.map.addControl(new mapboxgl.NavigationControl());
+
+      this.map.on("load", this.onMapLoad.bind(this));
 
-      // bind event handlers
-      map.on("load", this.onMapLoad.bind(this));
-      map.on("mousemove", "africa", this.onMapMouseMove.bind(this));
-      map.on("mouseleave", "africa", this.onMapMouseLeave.bind(this));
-      map.on("mousemove", "australia", this.onMapMouseMove.bind(this));
-      map.on("mouseleave", "australia", this.onMapMouseLeave.bind(this));
+      this.map.on("mousemove", "africa", this.onMapMouseMove.bind(this));
+      this.map.on("mouseleave", "africa", this.onMapMouseLeave.bind(this));
+      this.map.on("mousemove", "australia", this.onMapMouseMove.bind(this));
+      this.map.on("mouseleave", "australia", this.onMapMouseLeave.bind(this));
 
-      // control events
-      this.$watch("selectedMonth", this.onFilterChange.bind(this));
+      this.$watch("selectedMonth", this.filterBy.bind(this));
       this.$watch("selectedLayer", this.onBasemapChange.bind(this));
     },
 
-    // methods
+    onMapClick(e) {
+      if (e.features.length < 1) {
+        return;
+      }
+
+      this.map.getCanvas().style.cursor = "pointer";
+      this.props = JSON.parse(JSON.stringify(e.features[0].properties));
+    },
+
     filterBy(month) {
-      // set prediction filters
       const filter = ["==", "ts_month", month];
-      map.setFilter("africa", filter);
-      map.setFilter("australia", filter);
+      this.map.setFilter("africa", filter);
+      this.map.setFilter("australia", filter);
 
-      // set basemap
-      if (this.selectedLayer !== "esri_ocean") {
-        // get currently active basemap
-        const basemap = BASEMAPS.find((x) => x.key === this.selectedLayer);
+      if (this.selectedLayer === "esri_ocean") {
+        this.monthText = MONTHS[month - 1];
```

Comment:
This changes the original behavior. Now the month text will not be updated when the selected basemap is not ESRI. Bug!